relay-general/src/protocol/measurements.rs

```diff
@@ -0,0 +1,199 @@
+use regex::Regex;
+
+use crate::types::{Annotated, Error, FromValue, Object, Value};
+
+lazy_static::lazy_static! {
+    static ref MEASUREMENT_NAME: Regex = Regex::new("^[a-z0-9-._]+$").unwrap();
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, ToValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Measurement {
+    /// Value of observed measurement value
+    #[metastructure(required = "true")]
+    pub value: Annotated<f64>,
+}
+
+// TODO: need bag_size?
```

Comment:
Bag size is mostly useful for trimming, but we do not want to trim measurements. So, in this case we can go without.